ihatemoney/currency_convertor.py

```diff
@@ -14,205 +21,93 @@ def __call__(cls, *args, **kwargs):
         return cls._instances[cls]
 
 
-class CurrencyConverter(object, metaclass=Singleton):
-    # Get exchange rates
-    no_currency = "XXX"
+class ExchangeRateGetter(ABC):
+
+    def get_rates(self) -> Optional[ExchangeRates]:
+        """Method to retrieve a list of currency conversion rates.
+
+        Returns:
+            currencies: dict - key is a three-letter currency, value is float of conversion to base currency
+        """
+        try:
+            return self._get_rates()
+        except Exception:
+            warnings.warn(
+                f"Exchange rate getter failed - {traceback.format_exc(limit=0).strip()}"
+            )
+
+    @abstractmethod
+    def _get_rates(self) -> Optional[ExchangeRates]:
+        """Actual implementation of the exchange rate getter."""
+        raise NotImplementedError
+
+
+class ApiExchangeRate(ExchangeRateGetter):
     api_url = "https://api.exchangerate.host/latest?base=USD"
 
+    def _get_rates(self) -> Optional[ExchangeRates]:
+        return requests.get(self.api_url).json()["rates"]  # TODO not working currently probably
+
+
+class UserExchangeRate(ExchangeRateGetter):
+    user_csv_file = "path/to/file.csv"
```

Comment:
I don't believe using a CSV file would be useful here, instead, we could propose the user to define the rates themselves for the project, in the settings.

Reply:
Understandable. Makes sense. Will try to incorporate the settings :)